unreal/src/lib.rs

```diff
@@ -194,6 +223,66 @@ impl Unreal4Crash {
     pub fn get_context(&self) -> Result<Option<Unreal4Context>, Unreal4Error> {
         Unreal4Context::from_crash(self)
     }
+
+    /// Get up to `limit` log entries of this crash.
+    pub fn get_logs(&self, limit: usize) -> Result<Vec<Unreal4LogEntry>, Unreal4Error> {
+        match self.get_file_slice(Unreal4FileType::Log)? {
+            Some(f) => parse_log_from_slice(f, limit),
+            None => Ok(Vec::new()),
+        }
+    }
+}
+
+fn parse_log_from_slice(
+    log_slice: &[u8],
+    limit: usize,
+) -> Result<Vec<Unreal4LogEntry>, Unreal4Error> {
+    let mut fallback_timestamp = None;
+    let logs_utf8 = std::str::from_utf8(log_slice).map_err(Unreal4Error::InvalidLogEntry)?;
+
+    if let Some(first_line) = logs_utf8.lines().next() {
+        // First line includes the timestamp of the following 100 and some lines until
+        // log entries actually include timestamps
+        if let Some(captures) = LOG_FIRST_LINE.captures(&first_line) {
+            fallback_timestamp = Some(
+                // Using UTC but this entry is local time. Unfortunately there's no way to find the offset.
+                Utc.ymd(
+                    // https://github.com/EpicGames/UnrealEngine/blob/f7626ddd147fe20a6144b521a26739c863546f4a/Engine/Source/Runtime/Core/Private/GenericPlatform/GenericPlatformTime.cpp#L46
+                    captures.index("year").parse::<i32>().unwrap() + 2000,
+                    captures.index("month").parse::<u32>().unwrap(),
+                    captures.index("day").parse::<u32>().unwrap(),
```

Comment:
Use either `captures["year"].parse()` (and remove the import to `std::ops::Index`) or use `captures.name("year").unwrap().parse()`.